qdrant_docs_mcp/tools/importer.py

```diff
@@ -185,18 +185,23 @@ def extract_from_repo(
     """
 
     snippets: list[PartialSnippet] = []
-    paths = list(
-        chain(repo.glob("**/*.md"), repo.glob("**/*.rst"), repo.glob("**/*.ipynb"))
-    )
 
     if source.include_files is not None:
         filtered_paths: set[Path] = set()
         for pattern in source.include_files:
             filtered_paths |= {
-                path for path in paths if path.full_match(repo / pattern)
+                path
+                for path in repo.glob(pattern)
+                if path.full_match("**/*.md")
+                or path.full_match("**/*.rst")
+                or path.full_match("**/*.ipynb")
```

Comment:
is it possible to compare repo / pattern against "**/*.md", "**/*.rst", etc?
otherwise, we're doing it for each path

Reply:
That would not work for patterns like `my/docs/dir/**/*`.

Reply:
Also, the performance hit here isn't so bad. By far the slowest parts are the git clone and the upsert, so I wouldn't worry about this. And if we worry, I'd first put it all in multiprocessing and then think about further performance improvements.